Userland/DevTools/HackStudio/ToDoEntriesWidget.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2021, Federico Guerinoni <guerinoni.federico@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ToDoEntriesWidget.h"
+#include "HackStudio.h"
+#include "Project.h"
+#include "ToDoEntries.h"
+#include <AK/StringBuilder.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/TableView.h>
+#include <LibGUI/TextBox.h>
+#include <LibGfx/FontDatabase.h>
+
+namespace HackStudio {
+
+class ToDoEntriesModel final : public GUI::Model {
+public:
+    enum Column {
+        Filename,
+        Text,
+        __Count
+    };
+
+    explicit ToDoEntriesModel(const Vector<ToDoEntryPair>&& matches)
+        : m_matches(move(matches))
+    {
+    }
+
+    virtual int row_count(const GUI::ModelIndex& = GUI::ModelIndex()) const override { return m_matches.size(); }
+    virtual int column_count(const GUI::ModelIndex& = GUI::ModelIndex()) const override { return Column::__Count; }
+
+    virtual String column_name(int column) const override
+    {
+        switch (column) {
+        case Column::Filename:
+            return "Filename";
+        case Column::Text:
+            return "Text";
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    }
+
+    virtual GUI::Variant data(const GUI::ModelIndex& index, GUI::ModelRole role) const override
+    {
+        if (role == GUI::ModelRole::TextAlignment)
+            return Gfx::TextAlignment::CenterLeft;
+        if (role == GUI::ModelRole::Font) {
+            if (index.column() == Column::Text)
+                return Gfx::FontDatabase::default_fixed_width_font();
+            return {};
+        }
+        if (role == GUI::ModelRole::Display) {
+            auto& match = m_matches.at(index.row());
+            switch (index.column()) {
+            case Column::Filename:
+                return match.filename;
+            case Column::Text:
+                return match.comment;
```

Comment:
Not knowing the source position of a Todo/FIXME kinda just defeats the whole purpose of a widget like this
There would be lots of context-dependent to-dos without their context, and no way to get to it...

Reply:
Maybe with line can be useful? Or click to bring the cursor in that position...

Reply:
Yep.
I think LibCpp already knows about token positions, so you can probably just use those to calculate the positions for the TODOs/FIXMEs.

Reply:
Now I pass a `Vector<String>` through the functions, maybe I need some struct of TODO entry to pass

Reply:
Add position and click to bring the cursor on that TODO